internal/cmd/errors_userfacing.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+// UserFacingError wraps an error to provide better error messages to users.
+// Any non-UserFacingError will be printed as "unhandled errors".
+type UserFacingError struct {
+	// original error
+	Underlying error
+
+	// formatted outputs that users see when running the CLI commands
+	// should be a readable sentence rather than a stacktrace
+	UserFacingMessage string
+
+	// show or hide Underlying error from outputting with the UserFacingMessage
+	ShowUnderlying bool
+}
+
+func (u UserFacingError) Error() string {
```

Comment:
This seems unnecessarily complex. It should probably output the Message if it exists or the raw Error if it does not

Reply:
the reason I am attempting to make it a bit more complex, is so that user can have a general guideline of why that error is relevant to them, but still keep the original error.

First point (for context) is on why we need a user facing message at all (this was brought up a lot of times): 
- if I'm logging in, and it says connection refused, is this the infra server refusing connection or the provider? 
- if I'm adding a grant, why is there a random error saying "provider could not be added"?

Second point is then why do we need the original error:
- I initially suggested without it, that we either print `Internal error: stack trace...`, or a user facing message `This sis a sentence. To output a better sentence, add a provider. 'infra provider add PROVIDER'`
- however, after asking a couple of people, it seemed that a bunch of people would prefer to see the actual failure as well as a friendly error message